### PR TITLE
Update API types comments to replicate v1alpha2 comments

### PR DESCRIPTION
**Pull request by maelk**

### CODE REVIEW

File: api/v1alpha3/baremetalcluster_types.go [hunk 1]
```diff
@@ -90,6 +90,7 @@ type BareMetalClusterStatus struct {
 // +kubebuilder:printcolumn:name="Ready",type="string",JSONPath=".status.ready",description="BaremetalCluster is Ready"
 // +kubebuilder:printcolumn:name="Error",type="string",JSONPath=".status.failureReason",description="Most recent error"
 // +kubebuilder:printcolumn:name="Cluster",type="string",JSONPath=".metadata.labels.cluster\\.x-k8s\\.io/cluster-name",description="Cluster to which this BMCluster belongs"
+// +kubebuilder:printcolumn:name="APIEndpoints",type="string",JSONPath=".spec.controlPlaneEndpoint",description="API endpoint"
```
Comment (Xenwar) [inline api/v1alpha3/baremetalcluster_types.go]: The name is plural while the JSONPath='...' refers to a single value. 
Is there a mismatch ?

File: api/v1alpha3/baremetalmachine_types.go [hunk 2]
```diff
@@ -25,7 +25,7 @@ import (
 )
 
 const (
-	// MachineFinalizer allows ReconcileBareMetalMachine to clean up resources associated with AWSMachine before
+	// MachineFinalizer allows ReconcileBareMetalMachine to clean up resources associated with BareMetalMachine before
 	// removing it from the apiserver.
```
Comment (Xenwar) [inline api/v1alpha3/baremetalmachine_types.go:29]: from etcd ?

File: config/crd/bases/infrastructure.cluster.x-k8s.io_baremetalclusters.yaml [hunk 3]
```diff
@@ -142,6 +142,10 @@ spec:
       description: Cluster to which this BMCluster belongs
       name: Cluster
       type: string
+    - JSONPath: .spec.controlPlaneEndpoint
+      description: API endpoint
+      name: APIEndpoints
```
Comment (kashifest) [inline config/crd/bases/infrastructure.cluster.x-k8s.io_baremetalclusters.yaml]: ContolPlane Endpoint may be to be consistent with the field name? I would also add a few more words in description.

File: api/v1alpha3/baremetalmachine_types.go (hunk 2 above)
Comment (maelk) [reply api/v1alpha3/baremetalmachine_types.go:29]: it's actually interacting with api server to do the deletion, the client does not go directly to etcd.

File: api/v1alpha3/baremetalcluster_types.go (hunk 1 above)
Comment (maelk) [reply api/v1alpha3/baremetalcluster_types.go]: done

File: config/crd/bases/infrastructure.cluster.x-k8s.io_baremetalclusters.yaml (hunk 3 above)
Comment (maelk) [reply config/crd/bases/infrastructure.cluster.x-k8s.io_baremetalclusters.yaml]: done

**Review by kashifest — APPROVED** [review]

**Pull request merged**